src/posit/connect/system.py

```diff
@@ -0,0 +1,291 @@
+    ) -> None:
+        super().__init__(ctx, path, **attributes)
+
+    class _DestroyAttrs(TypedDict, total=False):
+        dry_run: bool
+        """If `True`, the cache will not be destroyed, only the operation will be simulated."""
+
+    @overload
+    def destroy(self, *, dry_run: Literal[True]) -> None: ...
+    @overload
+    def destroy(self, *, dry_run: Literal[False] = False) -> Task: ...
+    def destroy(self, **kwargs) -> Task | None:
+        """
+        Remove a content runtime package cache.
+
+        This action is only available to administrators.
+
+        Parameters
+        ----------
+        dry_run : bool, optional
+            If `True`, the cache will not be destroyed, only the operation will be simulated.
+
+        Returns
+        -------
+        Task | None
+            The task object if the operation was successful. If `dry_run=True`, `None` is returned.
+
+        Examples
+        --------
+        ```python
+        from posit.connect import Client
+
+        client = Client()
+
+        runtime_caches = client.system.caches.runtime.find()
+        first_runtime_cache = runtime_caches[0]
+
+        # Remove the cache
+        task = first_runtime_cache.destroy(dry_run=False)
+
+        # Wait for the task to finish
+        task.wait_for()
+        ```
+        """
+        url = self._ctx.url + self._path
+        data = dict(self)
+        response = self._ctx.session.delete(url, json={**data, **kwargs})
+
+        task_id = response.json().get("task_id")
+        if not task_id:
+            return None
+        task = self._ctx.client.tasks.get(task_id)
+        return task
+
+
+class SystemRuntimeCaches(ContextManager):
+    """
+    System runtime caches.
+
+    List all content runtime caches. These include packrat and Python
+    environment caches.
+
+    This information is available only to administrators.
+    """
+
+    def __init__(self, ctx: Context, path: str) -> None:
+        super().__init__()
+        self._ctx: Context = ctx
+        # v1/system/caches/runtime
+        self._path: str = path
+
+    def find(self) -> List[SystemRuntimeCache]:
+        """
+        List all content runtime caches.
+
+        List all content runtime caches. These include packrat and Python
+        environment caches.
+
+        This information is available only to administrators.
+
+        Returns
+        -------
+        List[SystemRuntimeCache]
+            List of all content runtime caches.
+
+        Examples
+        --------
+        ```python
+        from posit.connect import Client
+
+        client = Client()
+
+        runtime_caches = client.system.caches.runtime.find()
+        ```
+        """
+        url = self._ctx.url + self._path
+        response = self._ctx.session.get(url)
+        results = response.json()
+        if not isinstance(results, dict) and "caches" not in results:
+            raise RuntimeError("`caches=` not found in response.")
```

Comment:
Is there a known scenario where `caches` doesn't exist? On a fresh server? Otherwise, this check isn't necessary. This would indicate that something else went wrong, which should result in a server error.